sdk/storage/azure-storage-file-datalake/src/main/java/com/azure/storage/file/datalake/DataLakeDirectoryAsyncClient.java

```diff
@@ -0,0 +1,469 @@
+
+import static com.azure.core.implementation.util.FluxUtil.monoError;
+import static com.azure.core.implementation.util.FluxUtil.withContext;
+
+/**
+ * This class provides a client that contains directory operations for Azure Storage Data Lake. Operations provided by
+ * this client include creating a directory, deleting a directory, renaming a directory, setting metadata and
+ * http headers, setting and retrieving access control, getting properties and creating and deleting files and
+ * subdirectories.
+ *
+ * <p>
+ * This client is instantiated through {@link DataLakePathClientBuilder} or retrieved via
+ * {@link FileSystemAsyncClient#getDirectoryAsyncClient(String) getDirectoryAsyncClient}.
+ *
+ * <p>
+ * Please refer to the <a href=https://docs.microsoft.com/en-us/azure/storage/blobs/data-lake-storage-introduction?toc=%2fazure%2fstorage%2fblobs%2ftoc.json>Azure
+ * Docs</a> for more information.
+ */
+public final class DataLakeDirectoryAsyncClient extends DataLakePathAsyncClient {
+
+    private final ClientLogger logger = new ClientLogger(DataLakeDirectoryAsyncClient.class);
+
+    /**
+     * Package-private constructor for use by {@link DataLakePathClientBuilder}.
+     *
+     * @param pipeline The pipeline used to send and receive service requests.
+     * @param url The endpoint where to send service requests.
+     * @param serviceVersion The version of the service to receive requests.
+     * @param accountName The storage account name.
+     * @param fileSystemName The file system name.
+     * @param directoryName The directory name.
+     * @param blockBlobAsyncClient The underlying {@link BlobContainerAsyncClient}
+     */
+    DataLakeDirectoryAsyncClient(HttpPipeline pipeline, String url, DataLakeServiceVersion serviceVersion,
+        String accountName, String fileSystemName, String directoryName, BlockBlobAsyncClient blockBlobAsyncClient) {
+        super(pipeline, url, serviceVersion, accountName, fileSystemName, directoryName, blockBlobAsyncClient);
+    }
+
+    DataLakeDirectoryAsyncClient(DataLakePathAsyncClient dataLakePathAsyncClient) {
+        super(dataLakePathAsyncClient.getHttpPipeline(), dataLakePathAsyncClient.getPathUrl(), dataLakePathAsyncClient.getServiceVersion(),
+            dataLakePathAsyncClient.getAccountName(), dataLakePathAsyncClient.getFileSystemName(), dataLakePathAsyncClient.getObjectPath(),
+            dataLakePathAsyncClient.getBlockBlobAsyncClient());
+    }
+
+    /**
+     * Creates a directory.
+     *
+     * <p><strong>Code Samples</strong></p>
+     *
+     * {@codesnippet com.azure.storage.file.datalake.DataLakeDirectoryAsyncClient.create}
+     *
+     * <p>For more information see the
+     * <a href="https://docs.microsoft.com/en-us/rest/api/storageservices/datalakestoragegen2/path/create">Azure
+     * Docs</a></p>
+     *
+     * @return A reactive response containing information about the created directory.
+     */
+    public Mono<PathInfo> create() {
```

Comment:
Long term would this want to have a similar overload as blobs where a boolean could be passed to prevent overwriting?

Reply:
Issue for that created here https://github.com/Azure/azure-sdk-for-java/issues/6104